src/AdvancedVI.jl

```diff
@@ -210,61 +163,52 @@ Iteratively updates parameters by calling `grad!` and using the given `optimizer
 the steps.
 """
 function optimize!(
-    vo,
-    alg::VariationalInference,
-    q,
-    model,
-    θ::AbstractVector{<:Real};
-    optimizer = TruncatedADAGrad()
+    grad_estimator::AbstractGradientEstimator,
+    rebuild::Function,
```

Comment:
Generally, I'd recommend against forcing something like a `rebuild` method to be of type `Function`. This means that passing a callable `struct` won't be accepted even though it would technically be supported.

Reply:
This also applies in all the other places where you've specified `::Function`.

Reply:
Yes, I expect this to change in the future, depending on how we decide to flatten/deflatten the parameters.

Reply:
But IMO this should be changed no matter what because I might very well decide to instead do something like
```julia
struct ReBuilder end
(re::Rebuilder)(args...) = ...
```
and this isn't supported with `::Function` :confused:

Reply:
No problem. I'll rectify this next time.